discord/ui/select.py

```diff
@@ -836,11 +775,15 @@ async def select_channels(self, interaction: discord.Interaction, select: Channe
     def decorator(func: ItemCallbackType[V, BaseSelectT]) -> ItemCallbackType[V, BaseSelectT]:
         if not inspect.iscoroutinefunction(func):
             raise TypeError('select function must be a coroutine function')
-        if not issubclass(cls, BaseSelect):
+        if cls is MISSING:
+            callback_cls = _get_select_callback_parameter(func)
+        else:
+            callback_cls = getattr(cls, "__origin__", cls)
```

Comment:
What is this line for?

```suggestion
            callback_cls = getattr(cls, '__origin__', cls)
```

Reply:
It makes it so using `@ui.select(cls=ui.UserSelect[<type>])` wouldn't error anymore, but I can remove it if you want me to